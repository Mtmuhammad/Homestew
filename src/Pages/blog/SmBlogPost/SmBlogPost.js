import React from "react";
import "./SmBlogPost.scss";

const SmBlogPost = ({ img, title, category }) => {
  return (
    <div className="text-center mb-5 sm-blog-post col-12 col-md-6 col-lg-4">
      <img className="img-fluid" src={img} alt={title} />
      <div className="sm-blog-text text-light">
        <p role="paragraph" className="my-4">
          {category}
        </p>
        <h4>{title}</h4>
        <a href="#0">Read More</a>
      </div>
    </div>
  );
};

export default SmBlogPost;
