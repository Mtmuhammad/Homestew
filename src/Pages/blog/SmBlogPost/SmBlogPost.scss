@use "../../../Styles/abstracts/mixins" as *;
@use "../../../Styles/abstracts/variables" as *;

.blog {
  .fa-utensils {
    color: $primary-green;
    font-size: 3.8rem;
    margin-left: 1rem;
  }
}
.sm-blog {
  &-post {
    height: 30rem;
    position: relative;

    & img {
      width: 100%;
      height: 100%;
      box-sizing: border-box;
      background-size: cover;
      object-fit: cover;
      border-radius: 5px;
    }
  }
  &-text {
    background-color: rgba(0, 0, 0, 0.5);
    width: 95%;
    height: auto;
    position: absolute;
    bottom: 0;
    border-top: 1px solid $primary-green;
    padding: 2rem 0;

    & h4 {
      font-size: 2.2rem;
      margin-bottom: 2rem;

      @include respond(sport) {
        font-size: 3rem;
      }
    }

    & p {
      line-height: 2px;
      letter-spacing: 0.2rem;
      font-size: 1.6rem;
      text-transform: uppercase;
      color: #fff;

      @include respond(sport) {
        font-size: 2rem;
      }
    }

    & a {
      color: #fff;
      background-color: $primary-green;
      padding: 1rem;
      font-size: 1.2rem;
      text-transform: uppercase;
      border-radius: 5px;
      transition: all 0.3s;

      @include respond(sport) {
        font-size: 1.6rem;
      }

      &:hover {
        color: $primary-font;
      }
    }
  }
}
