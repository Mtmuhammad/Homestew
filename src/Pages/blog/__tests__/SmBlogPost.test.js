import React from "react";
import { render, screen } from "@testing-library/react";
import SmBlogPost from "../SmBlogPost/SmBlogPost";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(
      <SmBlogPost
        category={"travel"}
        img={"Sm1.jpg"}
        title={"The Travelers Guide for Foodies"}
      />
    );
  });

  test("should match snapshot", () => {
    const { asFragment } = render(
      <SmBlogPost
        category={"travel"}
        img={"Sm1.jpg"}
        title={"The Travelers Guide for Foodies"}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

test("should render img", () => {
  render(
    <SmBlogPost
      category={"travel"}
      img={"Sm1.jpg"}
      title={"The Travelers Guide for Foodies"}
    />
  );

  const img = screen.getByAltText(/the travelers guide for foodies/i);
  expect(img).toBeInTheDocument();
  expect(img).toHaveClass("img-fluid");
});

test("should render category", () => {
  render(
    <SmBlogPost
      category={"travel"}
      img={"Sm1.jpg"}
      title={"The Travelers Guide for Foodies"}
    />
  );

  const category = screen.getByRole("paragraph");
  expect(category).toBeInTheDocument();
  expect(category).toHaveTextContent(/travel/i);
});

test("should render heading", () => {
  render(
    <SmBlogPost
      category={"travel"}
      img={"Sm1.jpg"}
      title={"The Travelers Guide for Foodies"}
    />
  );

  const category = screen.getByRole("heading");
  expect(category).toBeInTheDocument();
  expect(category).toHaveTextContent(/the travelers guide for foodies/i);
});
