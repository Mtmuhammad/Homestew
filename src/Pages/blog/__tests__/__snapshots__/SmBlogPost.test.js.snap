// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center mb-5 sm-blog-post col-12 col-md-6 col-lg-4"
  >
    <img
      alt="The Travelers Guide for Foodies"
      class="img-fluid"
      src="Sm1.jpg"
    />
    <div
      class="sm-blog-text text-light"
    >
      <p
        class="my-4"
        role="paragraph"
      >
        travel
      </p>
      <h4>
        The Travelers Guide for Foodies
      </h4>
      <a
        href="#0"
      >
        Read More
      </a>
    </div>
  </div>
</DocumentFragment>
`;
